lua/modules/particles.lua

```diff
@@ -60,20 +65,33 @@ particles.newEmitter = function(_, config)
 			p.Physics = particles.fromConfigOrDefault(conf.physics, true)
 			p.Velocity = particles.fromConfigOrDefault(conf.velocity, Number3(0, 0, 0))
 			p.Position = self.Position + particles.fromConfigOrDefault(conf.position, Number3(0, 0, 0))
+			p.Rotation = particles.fromConfigOrDefault(conf.rotation, Rotation(0, 0, 0))
 			p.Acceleration = particles.fromConfigOrDefault(conf.acceleration, Number3(0, 0, 0))
-			p.Scale = particles.fromConfigOrDefault(conf.scale, 1.0)
+			p.startScale = particles.fromConfigOrDefault(conf.start_scale, particles.fromConfigOrDefault(conf.scale, 1.0))
+			p.endScale = particles.fromConfigOrDefault(conf.end_scale, particles.fromConfigOrDefault(conf.start_scale, particles.fromConfigOrDefault(conf.scale, 1.0)))
 			p.Mass = particles.fromConfigOrDefault(conf.mass, 1.0)
 			p.Palette[1].Color = particles.fromConfigOrDefault(conf.color, Color.White)
 			p.CollisionGroups = particles.fromConfigOrDefault(conf.collisionGroups, nil)
 			p.CollidesWithGroups = particles.fromConfigOrDefault(conf.collidesWithGroups, nil)
+			p.startLife = particles.fromConfigOrDefault(conf.life, 10.0)
+			p.life = particles.fromConfigOrDefault(conf.life, 10.0)
+
+			p.remove = function(s)
+				s.Tick = nil
+				s:RemoveFromParent()
+				table.insert(particles.pool, s)
+			end
 
 			World:AddChild(p)
 
-			local life = particles.fromConfigOrDefault(conf.life, 10.0)
-			Timer(life, function()
-				p:RemoveFromParent()
-				table.insert(particles.pool, p)
-			end)
+			p.Tick = function(s, dt)
+				s.Scale = particles.lerp(s.startScale, s.endScale, 1-(s.life/s.startLife))
+				
+				s.life = s.life - dt
+				if s.life <= 0 then
+					s:remove()
+				end
+			end
```

Comment:
It's not ideal to instantiate one Tick for each particle. 
Actually, Timers weren't ideal even. 
Now that we have local events, the module should listen for tick events. 
And each particle could carry its creation date `Time:UnixMilli()`. 
Within one single tick, we could compare with `now` to learn and remove when particle's life is over.

Reply:
i think it will be worse that this system